### PR TITLE
feat: Add source attribution reporting on DocumentIndexingPipeline + Tests

### PR DESCRIPTION
**Pull request by eob**
Right now the document indexing pipeline does not properly include source material attribution.

This PR adds and tests that inclusion. The source attribution is both automatic (fileId, blockId, page) and user-controlled (via the metadata argument passed in at index request time)

### CODE REVIEW

File: tests/steamship_tests/app/integration/test_e2e_mixins_indexer_pipeline.py [hunk 1]
```diff
@@ -28,6 +28,38 @@ def test_indexer_pipeline_mixin(client: Steamship):
...
+        assert winner.tag.value.get("block_id")  # It has a block id stamped
+        assert winner.tag.value.get("file_id")  # It has a file id stamped
+        assert winner.tag.value.get("page") == 0
+
+        # LOAD THE TAO TE QING PDF
+        # This will test metadata
+        pdf_url2 = "https://www.with.org/tao_te_ching_en.pdf"
+
+        index_task2 = instance.invoke(
+            "index_url", url=pdf_url2, metadata={"url": pdf_url2, "is_tao": True}
```
[the remark this answers is not in the document]
Comment (eob) [reply tests/steamship_tests/app/integration/test_e2e_mixins_indexer_pipeline.py:41]: Yeah that's a good idea -- I'll make that update.

[the remark this answers is not in the document]
Comment (eob) [reply tests/steamship_tests/app/integration/test_e2e_mixins_indexer_pipeline.py:41]: Also adding in the mime type of the containing file

File: src/steamship/agents/tools/question_answering/vector_search_qa_tool.py [hunk 2]
```diff
@@ -43,17 +44,31 @@ def answer_question(self, question: str, context: AgentContext) -> List[Block]:
...
 
         final_prompt = self.question_answering_prompt.format(
             **{"source_text": "\n".join(source_texts), "question": question}
         )
 
-        return get_llm(context, default=OpenAI(client=context.client)).complete(prompt=final_prompt)
+        output_blocks = get_llm(context, default=OpenAI(client=context.client)).complete(
+            prompt=final_prompt
+        )
+        for output_block in output_blocks:
```
Comment (douglas-reid) [inline src/steamship/agents/tools/question_answering/vector_search_qa_tool.py:65]: Looking at this again, i wonder if this isn't slightly cheating / too clever -- The LLM isn't telling us which sources it used, right?  So, we are just saying which sources we found and added to the query. But what if the LLM ignored the sources and/or only used certain ones?

Comment (eob) [reply src/steamship/agents/tools/question_answering/vector_search_qa_tool.py:65]: I'm almost positive I've seen some prompts ask the LLM to use citations for exactly this reason, though I don't know how well it works -- I can look into that and do a few test runs to see how stable the output is.

**Review by douglas-reid — APPROVED** [review]
I think this is good enough to start with. We can refine as needed.

**Pull request merged**